### ISSUE TITLE
`:set virtualedit=all` not working when called on a `:term` through RPC

### ISSUE THREAD

**Issue by glacambre** [neovim/neovim#8243]
- `nvim --version`: v0.2.3-969-g2ec94eb23
- Operating system/version: Linux 4.14.0-3-amd64 #1 SMP Debian
- Terminal name/version: rxvt-unicode (urxvt) v9.22
- `$TERM`: rxvt-unicode-256color

### Steps to reproduce
Use the following init.vim:
```vim
function! GetTermWidth()
	if mode("") != "n"
		call nvim_input("<c-\\><c-n>")
	endif
	set virtualedit=all
	call nvim_input("g$")
	let g:width = virtcol('.')
	return g:width
endfunction

term sh -c echo "" && python -c "$(echo -ne 'import os\n
			\import neovim\n
			\width = neovim.attach("socket",path=os.environ.get("NVIM_LISTEN_ADDRESS",None)).funcs.GetTermWidth()\n
			\print(width)\n')"

```
And then run nvim -u init.vim

### Actual behaviour
The terminal buffer displays `1` and `g:width` is set to the same value.

### Expected behaviour
The terminal buffer should display the width of the window.

Calling `GetTermWidth()` directly from neovim seems to work. Adding a delay (e.g. `sleep 1`) before `python -c` doesn't change anything. The reason why I'm using `virtualedit` and `g$` is that I don't want the width of the number column/gutter/whatever other GUI element there could be to be taken into account. Please let me know if there's a better solution.

This issue seems a bit similar to https://github.com/neovim/neovim/issues/7807 .

**Comment by glacambre** [comment]
@justinmk I see your labeled this issue as `complexity:low`. I'd like to try to fix this issue, do you have any idea where I should start looking?

**Comment by justinmk** [comment]
I don't think this has anything to do with `virtualedit`. You can even `:set virtualedit=all` manually, but the result of `virtcol('.')` is still 1. 

`nvim_input()` is asynchronous, it only sends input. There's no guarantee about _when_ that input is processed, and certainly it may not be processed until after `GetTermWidth()` returns.

If you use `normal! g$`  instead of `nvim_input('g$')` then it works.

**Comment by glacambre** [comment]
>nvim_input() is asynchronous
So, is this behavior actually expected? Should I close this issue?
